assets/js/stripe.js

```diff
@@ -398,305 +508,281 @@ jQuery( function( $ ) {
+		 *
+		 * @param {Object} response The `stripe.createSource` response.
+		 */
 		sourceResponse: function( response ) {
-			if ( response.error ) {
-				$( document.body ).trigger( 'stripeError', response );
-			} else if ( 'no' === wc_stripe_params.allow_prepaid_card && 'card' === response.source.type && 'prepaid' === response.source.card.funding ) {
-				response.error = { message: wc_stripe_params.no_prepaid_card_msg };
-
-				if ( 'yes' === wc_stripe_params.is_stripe_checkout ) {
-					wc_stripe_form.submitError( '<ul class="woocommerce-error"><li>' + wc_stripe_params.no_prepaid_card_msg + '</li></ul>' );
-				} else {
-					$( document.body ).trigger( 'stripeError', response );
-				}
-			} else {
-				wc_stripe_form.processStripeResponse( response.source );
-			}
-		},
-
-		intentResponse: function( response ) {
 			if ( response.error ) {
 				return $( document.body ).trigger( 'stripeError', response );
 			}
 
-			var intent = response.paymentIntent;
-
 			wc_stripe_form.reset();
-			wc_stripe_form.form.append( "<input type='hidden' class='stripe-intent' name='stripe_intent' value='" + intent.id + "'/>" );
+			wc_stripe_form.addHiddenInput( 'source', response.source.id );
 
-			// ToDo: Check for `form#add_payment_method` and remove event listeners
+			if ( $( 'form#add_payment_method' ).length ) {
+				$( wc_stripe_form.form ).off( 'submit', wc_stripe_form.form.onSubmit );
+			}
 
 			wc_stripe_form.form.submit();
 		},
 
-		processStripeResponse: function( source ) {
-			wc_stripe_form.reset();
+		/**
+		 * Responds to payments with credit cards, which use PaymentIntents.
+		 *
+		 * @param {object} response The response from calling `stripe.handleCardPayment`.
+		 */
+		paymentIntentResponse: function( response ) {
+			if ( response.error ) {
+				return $( document.body ).trigger( 'stripeError', response );
+			}
 
-			// Insert the Source into the form so it gets submitted to the server.
-			wc_stripe_form.form.append( "<input type='hidden' class='stripe-source' name='stripe_source' value='" + source.id + "'/>" );
+			wc_stripe_form.reset();
+			wc_stripe_form.addHiddenInput( 'intent', response.paymentIntent.id );
 
-			if ( $( 'form#add_payment_method' ).length ) {
-				$( wc_stripe_form.form ).off( 'submit', wc_stripe_form.form.onSubmit );
-			}
+			// ToDo: Check for `form#add_payment_method` and remove event listeners
 
 			wc_stripe_form.form.submit();
 		},
 
+		/**
+		 * Performs payment-related actions when a checkout/payment form is being submitted.
+		 *
+		 * @param {Event} e A form submission event.
+		 */
 		onSubmit: function( e ) {
 			if ( ! wc_stripe_form.isStripeChosen() ) {
 				return;
 			}
 
-			if ( ! wc_stripe_form.isStripeSaveCardChosen() && ! wc_stripe_form.hasSource() && ! wc_stripe_form.hasToken() && ! wc_stripe_form.hasIntent() ) {
-				e.preventDefault();
+			/**
+			 * ToDo: The logic here was restructured a bit, make sure it still works well.
+			 */
 
-				wc_stripe_form.block();
```

Comment:
In the new logic, there are a lot of cases where you aren't calling `wc_stripe_form.block();`, I assume that's gonna cause some problems.

Reply:
Actually, it's being removed in two places:

1. Here, where I simply seem to have missed it. It's back now.
2. The `form#add_payment_method` form section of `onSubmit`, which is now removed.

Now it should be in sufficiently many places :)